opencog/rule-engine/backwardchainer/BackwardChainer.cc

```diff
@@ -197,7 +197,17 @@ void BackwardChainer::fulfill_fcs(const Handle& fcs)
 	// results
 	AtomSpace tmp_as(&_as);
 
-	// Run the FCS and add the results in _as
+	// Run the FCS and add the results, if any, in _as.
+	//
+	// Warning: since tmp_as is a child of _as, TVs of existing atoms
+	// in _as, that are modified by running fcs will be modified on
+	// _as as well. This can create involontary TVs changes, hopefully
+	// mitigated by the merging the TVs properly (for now the one with
+	// the highest confidence wins). To avoid that side effect, we
+	// could operate on a copy the atomspace, of its zone of focus. Or
+	// alternatively modify some HypotheticalLink wrapping the atoms
+	// of concerns instead of the atoms themselves, and only modify
+	// the atoms if there are existing results to copy back to _as.
 	Handle hresult = bindlink(&tmp_as, fcs);
```

Comment:
This is a very interesting comment, and it's what I was talking about a year or two ago about "kripke frames". So, as the chainer is running, you want to have some way of making it look as if there were multiple atomspaces, each one holding possibly different TV's for the same atom.  The TV's in the "main" atomspace would only be updated (maybe) at the very end of the chaining.

I was calling each of these different atomspaces "parallel universes". 

The comment above seems like you are struggling with exactly this idea. 

The actual implementation of this idea does not actually have to use different atomspaces .. any technique where you can have one atom with different TV's on them will be enough.